Movied/Movied/Scenes/MovieDetail/MovieDetailViewController.swift

```diff
@@ -0,0 +1,173 @@
+import UIKit
+
+final class MovieDetailViewController: UIViewController {
+    
+    @IBOutlet private weak var bookNowButton: UIButton!
+    @IBOutlet private weak var movieBackgroundImage: UIImageView!
+    @IBOutlet private weak var moviePosterImage: UIImageView!
+    @IBOutlet private var movieRateStarImageViews: [UIImageView]!
+    @IBOutlet private weak var movieNameLabel: UILabel!
+    @IBOutlet private weak var movieRateLabel: UILabel!
+    @IBOutlet private weak var movieLanguageLabel: UILabel!
+    @IBOutlet private weak var movieDurationLabel: UILabel!
+    @IBOutlet private weak var movieSynopsisLabel: UILabel!
+    @IBOutlet private weak var genresCollectionView: UICollectionView!
+    @IBOutlet private weak var actorsCollectionView: UICollectionView!
+    
+    private let dispatchGroup = DispatchGroup()
+    
+    var movieId: Int?
+    var movieDetail: MovieDetail?
+    var movieActors: [Actor] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configView()
+        fetchMovieInfoData()
+    }
+    
+    private func fetchMovieInfoData() {
+        showIndicator(true)
+        dispatchGroup.enter()
+        APIService.shared.getMovieDetail(id: movieId ?? 0) { [weak self] result in
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                switch result {
+                case .success(let data):
+                    if let movieDetail = data {
+                        self.movieDetail = movieDetail
+                        self.fillData(with: movieDetail)
+                        self.genresCollectionView.reloadData()
+                    }
+                case .failure(let error):
+                    print(error.rawValue)
+                }
+            }
+        }
+        dispatchGroup.leave()
```

Comment:
Need to research more about DispatchGroup, when call enter(), when call leave() and the count of enter() and leave(), and why do that?
Here, please move this command inside call back block code of APIService.shared.getMovieDetail, to inform to DispatchGroup that the task is done, because this command will be executed immediately then APIService.shared.getMovieDetail, so it's not really the getMovieDetail is completed, difference queue.